JetNews/app/src/main/java/com/example/jetnews/ui/home/HomeScreens.kt

```diff
@@ -60,76 +66,176 @@ import com.example.jetnews.R
- * @param openDrawer (event) request opening the app drawer
- * @param scaffoldState (state) state for the [Scaffold] component on this screen
+ * The home screen displaying a list and a detail.
  */
 @Composable
-fun HomeScreen(
-    homeViewModel: HomeViewModel,
-    navigateToArticle: (String) -> Unit,
+fun HomeListDetailScreen(
+    uiState: HomeUiState,
+    onToggleFavorite: (String) -> Unit,
+    onSelectPost: (String) -> Unit,
+    onRefreshPosts: () -> Unit,
+    onErrorDismiss: (Long) -> Unit,
+    onInteractWithList: () -> Unit,
+    onInteractWithDetail: (String) -> Unit,
     openDrawer: () -> Unit,
-    scaffoldState: ScaffoldState = rememberScaffoldState()
+    homeListLazyListState: LazyListState,
+    articleDetailLazyListStates: Map<String, LazyListState>,
+    scaffoldState: ScaffoldState
 ) {
-    // UiState of the HomeScreen
-    val uiState by homeViewModel.uiState.collectAsState()
+    HomeScreenWithList(
+        uiState = uiState,
+        onRefreshPosts = onRefreshPosts,
+        onErrorDismiss = onErrorDismiss,
+        openDrawer = openDrawer,
+        homeListLazyListState = homeListLazyListState,
+        scaffoldState = scaffoldState
+    ) { hasPostsUiState, modifier ->
+        Row {
+            PostList(
+                postsFeed = hasPostsUiState.postsFeed,
+                onArticleTapped = onSelectPost,
+                favorites = hasPostsUiState.favorites,
+                onToggleFavorite = onToggleFavorite,
+                contentPadding = rememberInsetsPaddingValues(
+                    insets = LocalWindowInsets.current.systemBars,
+                    applyTop = false,
+                    applyEnd = false,
+                ),
+                modifier = modifier
+                    .width(334.dp)
+                    .pointerInput(Unit) {
+                        while (currentCoroutineContext().isActive) {
+                            awaitPointerEventScope {
+                                awaitPointerEvent(PointerEventPass.Initial)
+                                onInteractWithList()
+                            }
+                        }
+                    },
+                state = homeListLazyListState
+            )
+            // Crossfade between different detail posts
+            Crossfade(targetState = hasPostsUiState.selectedPost) { detailPost ->
+                // Get the lazy list state for this detail view
+                val detailLazyListState by derivedStateOf {
+                    articleDetailLazyListStates.getValue(detailPost.id)
+                }
+
+                // Key against the post id to avoid sharing any state between different posts
+                key(detailPost.id) {
+                    PostContent(
+                        post = detailPost,
+                        modifier = modifier
+                            .fillMaxSize()
+                            .pointerInput(Unit) {
```

Comment:
The logic for this `pointerInput` and the above is the same except the action to take. We could create a modifier out of this with a more meaningful name and pass the lambda in

Reply:
We need to be conscious about this as every user input triggers a `uiState` update in the VM

Reply:
I created a modifier here, and removed one place where it was unnecessary.
I feel like we shouldn't need to be too concerned about this firing extremely often, the only expense should be contesting the CAS loop?